AutoOption.cmake

```diff
@@ -0,0 +1,45 @@
+# Add a new option with default value "auto":
+#  autoOption(FOOBAR "Support foobar")
+#
+# To check if FOOBAR support should be enabled, use:
+#   if(FOOBAR)
+#
+# If necessary dependencies for FOOBAR are missing, use:
+#   autoDisable(FOOBAR "Did not find baz")
+#
+# Example:
+#  autoOption(FOOBAR "Support foobar")
+#   if(FOOBAR)
+#     Check for FOOBAR here.
+#     if(NOT BAZ_FOUND)
+#       autoDisable(FOOBAR "Did not find baz")
+#     endif()
+#   endif()
+#
+# If necessary dependencies for FOOBAR are missing, use:
+#   autoDisable(FOOBAR "Did not find baz")
+
+function(autoOption name description)
+    set(${name} AUTO CACHE STRING "${description}")
+    set_property(CACHE ${name} PROPERTY STRINGS AUTO ON OFF)
+
+    if((NOT ${name} STREQUAL ON) AND
+        (NOT ${name} STREQUAL OFF) AND
+        (NOT ${name} STREQUAL AUTO))
+        message(FATAL_ERROR "Value of ${name} must be one of ON/OFF/AUTO, but is ${${name}}")
+    endif()
+endfunction()
+
+function(autoDisable name reason)
+    message(STATUS "${reason}")
```

Comment:
Should `name` be included in the status message?

Reply:
The two calls to this function have message `DBus not found.` and `Not generating manpages. Missing: [stuff]`. That seems quite self-explanatory to me and adding `WITH_DBUS` or `GENERATE_MANPAGES` in there won't really improve things, I think.

Reply:
Ok.